test/Triton/rewrite-tensor-pointer.mlir

```diff
@@ -81,3 +81,20 @@ tt.func public @matmul_kernel(%arg0: !tt.ptr<f16> {tt.divisibility = 16 : i32},
   tt.store %45, %30, %54 : tensor<128x32x!tt.ptr<f16>>
   tt.return
 }
+
+// -----
+
+tt.func public @asm_in_loop(%arg0: !tt.ptr<bf16> {tt.divisibility = 16 : i32}) attributes {noinline = false} {
+  %c0_i32 = arith.constant 0 : i32
+  %c1_i32 = arith.constant 1 : i32
+  %c0_i64 = arith.constant 0 : i64
+  %c128_i64 = arith.constant 128 : i64
+  %0 = tt.make_range {end = 16 : i32, start = 0 : i32} : tensor<16xi32>
+  %1 = tt.make_tensor_ptr %arg0, [%c128_i64, %c128_i64], [%c128_i64, %c0_i64], [%c0_i32, %c0_i32] {order = array<i32: 0, 1>} : <tensor<128x128xbf16>>
+  %2:1 = scf.for %arg1 = %c0_i32 to %c1_i32 step %c1_i32 iter_args(%arg2 = %1) -> (!tt.ptr<tensor<128x128xbf16>>)  : i32 {
+    %3:2 = tt.elementwise_inline_asm "{.reg .b8 tmp<4>; mov.b32 {tmp0, tmp1, tmp2, tmp3}, $2; cvt.u16.u8 $0, tmp0; cvt.u16.u8 $1, tmp1;}" {constraints = "=r,=r,r", packed_element = 1 : i32, pure = true} %0 : tensor<16xi32> -> tensor<16xi16>, tensor<16xi16>
```

Comment:
is this ops needed at all?

Reply:
While it may not be actually meaningful, it does trigger the bug.

Although it could be removed with all passes since the return value is not actually used, the test specifically designed with the appointed pass (-triton-rewrite-tensor-pointer) does indeed expose the bug.

Reply:
do you understand why it triggers the bug? This looks like it is a dead op.

Reply:
The bug was triggered because elementwise_inline_asm returned more results than scf.for. 

I encountered a bug while using inline_asm. I'm not sure which other simple operations return multiple results. As I am not very familiar with MLIR, could you suggest a good alternative?

Reply:
aah I see, thanks for explaining. Then maybe just replacing the string with something simple would make the ir look simpler and show what you mean to test. This is very much a nitpick though:
```
%3:2 = tt.elementwise_inline_asm "asm_multiple_results" {constraints = "=r,=r,r", packed_element = 1 : i32, pure = true} %0 : tensor<16xi32> -> tensor<16xi16>, tensor<16xi16>
```